persistence/src/test/scala/akka/persistence/LocalActorRefTest.scala

```diff
@@ -0,0 +1,101 @@
+package akka.persistence
+
+import cats.effect.IO
+import cats.effect.unsafe.implicits.global
+import cats.syntax.all._
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.funsuite.AnyFunSuite
+import scala.concurrent.duration._
+import java.util.concurrent.TimeoutException
+
+class LocalActorRefTest extends AnyFunSuite with Matchers {
```

Comment:
This test is using a lot of sleeps in it. I speculate it is going to be flaky in CI environment, and also may give false negatives under load (i.e. test pass, but there is a borderline bug).

As the implementation uses `Concurrent` etc., then the trivial approach of mocking `Clock` etc. won't work, and it might be worth to look into using Cats Effect Test Runtime: https://typelevel.org/cats-effect/docs/core/test-runtime